model/src/pyrenew/metaclass.py

```diff
@@ -152,6 +192,15 @@ def set_timeseries(
         -------
         None
         """
+
+        # Either both values are None or both are not None
+        assert (t_unit is not None and t_start is not None) or (
+            t_unit is None and t_start is None
+        ), "Both t_start and t_unit should be None or not None."
```

Comment:
I think we could be more helpful here by printing `t_start` and `t_unit` in the error message.